app/Http/Controllers/PublisherController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\PublisherResource;
+use App\Models\Publisher;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+
+class PublisherController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(Request $request)
+    {
+        try {
+            $perPageOptions = [20, 50, 100];
+
+            $searchValue = $request->input('search', '');
+            $perPage = $request->input('per_page', 20); // Default to 20 if not defined
+
+            if (!in_array($perPage, $perPageOptions)) {
```

Comment:
@damir-tahirovic It's better to validate per_page as well. This check can also be included in the validator.